games/game_bladeandsorcery.py

```diff
@@ -1,18 +1,198 @@
+from pathlib import Path
+import json
+import os
+from typing import List
+
+from PyQt6.QtCore import QDateTime, QDir, QLocale, Qt
+from PyQt6.QtGui import QFont
+
+import mobase
+
+from PyQt6.QtWidgets import QSizePolicy, QVBoxLayout, QFormLayout, QLabel, QStyle
+from ..basic_features.basic_save_game_info import (
+    BasicGameSaveGame,
+    BasicGameSaveGameInfo
+)
+
 from ..basic_game import BasicGame
 
 
-class BaSGame(BasicGame):
+class BaSSaveGame(BasicGameSaveGame):
+    def __init__(self, filepath):
+        super().__init__(filepath)
+        self._filepath = Path(filepath)
+        save = open(self._filepath, "rb")
```

Comment:
This should be done with `with ...` block rather than with a open then close:

```python
with open(self._filepath, "rb") as fp:
    save_data = json_load(save)
```